--- mpich-4.3.0/src/util/mpir_pmi.c	2025-02-03 09:07:01.000000000 -0600
+++ mpich-4.3.0-new/src/util/mpir_pmi.c	2025-07-07 17:02:07.797611725 -0500
@@ -149,6 +149,10 @@
                mpi_errno = pmix_init(&has_parent, &rank, &size, &appnum));
     MPIR_ERR_CHECK(mpi_errno);
 
+    /* allowing user to override default KEY VAL size for certain system setups */
+    MPL_env2int("PMI_MAX_KEY_SIZE", &pmi_max_key_size);
+    MPL_env2int("PMI_MAX_VAL_SIZE", &pmi_max_val_size);
+
     const char *hostname;
     if (MPL_env2str("PMI_HOSTNAME", &hostname)) {
         pmi_hostname = MPL_strdup(hostname);
